assets/app/main.js

```diff
@@ -8,48 +8,26 @@ import siteActions from './actions/siteActions';
 import userActions from './actions/userActions';
 
 import routes from './routes';
-
-import router from './router';
 import store from './store';
+import Provider from './util/provider';
 
 const mainEl = document.querySelector('#js-app');
-class Provider extends React.Component {
-  getChildContext() {
-    return { state: this.state }
-  }
-
-  constructor(props, context) {
-    super(props, context)
-    this.state = props.state
-  }
-
-  render() {
-    return React.Children.only(this.props.children)
-  }
-}
-
-Provider.childContextTypes = {
-  state: React.PropTypes.object
-};
 
 store.subscribe(() => {
   const state = store.getState();
+
   if (!!!state.user) return;
 
   render((
-    <Provider state={state}>
+    <Provider state={{get: store.getState}}>
```

Comment:
App.js needs to have access to the store.getState function so it can call it on componentWillReceiveProps and update its own internal state. We can also pass the output of store.getState() to app.js and just set state with the entire context

```
componentWillReceiveProps(nextProps, nextContext) {
  this.setState(nextContext);
}
```